lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2206,5 +2212,68 @@ def test_data_bool_not_eq(self):
         self.assertFalse(cube1 == cube2)
 
 
+class Test__eq__meta(tests.IrisTest):
+    def test_ancillary_fail(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        avr = AncillaryVariable([2, 3], long_name="foo")
+        cube2.add_ancillary_variable(avr, 0)
+        self.assertFalse(cube1 == cube2)
```

Comment:
I think it might be worth checking that the values in ancillaries count : I.e. cubes with the same ancillaries, but some values different, will not match.
You could extend that to attributes and units, if you want to be really thorough.

You may find, as I did in #3550, that some of this could be simpler + clearer if we can get [cube.copy support for ancillaries](https://github.com/SciTools/iris/pull/3546) in first.

Reply:
The tests in #3550, as I have written them, use equality. At the moment, those tests aren't working properly so there isn't a full level of confidence (with respect to travis at least) in that pull request. I'm not sure if having equality testing rely upon copy and copy testing rely upon equality is a great idea anyway, so I think at least one of those tests should be done the long way.